src/cryptography/x509.py

```diff
@@ -1318,6 +1322,11 @@ def __init__(self, key_identifier, authority_cert_issuer,
         self._authority_cert_issuer = authority_cert_issuer
         self._authority_cert_serial_number = authority_cert_serial_number
 
+    @classmethod
+    def from_issuer_certificate(cls, certificate):
+        digest = _key_identifier_from_public_key(certificate.public_key())
+        return cls(digest, None, None)
```

Comment:
Is there a reason we don't populate the issuer or serial number?

And if we're not going to, why not just have this take teh public key?